### PR TITLE
VideoPlayer: ffmpeg, fix calculating aspect ratio for 3d modes

### PR DESCRIPTION
**Pull request by FernetMenta**
finally fix http://trac.kodi.tv/ticket/16876

### CODE REVIEW

**Comment by afedchin** [comment]
thanks. will test this tonight.

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp [hunk 1]
```diff
@@ -833,7 +833,13 @@ bool CDVDVideoCodecFFmpeg::GetPictureCommon(DVDVideoPicture* pDvdVideoPicture)
     aspect_ratio = av_q2d(pixel_aspect) * pDvdVideoPicture->iWidth / pDvdVideoPicture->iHeight;
 
   if (aspect_ratio <= 0.0)
+  {
```
Comment (afedchin) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp]: If aspect is calculated already then all code below will not work. so the correction doesn't work also.

**Comment by afedchin** [comment]
I can't test with all my samples, but seems with this an anamorph video is displayed wrong. 
I'll try to explain:
1. for anamorph videos (1920x1080) we should use codec aspect without changes.
2. for non anamorph videos (1920x2160 or 3840x1080) we should change aspect to *2 or /2

**Comment by afedchin** [comment]
here is a sample https://www.dropbox.com/s/q5y3q4jee30y4hx/3d.sbs.disney_3d_global_sizzle_2012-DWEU.m2ts?dl=0 which doesn't work with this

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp (hunk 1 above)
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp]: this block is only executed if codec does not provide AR

**Comment by afedchin** [comment]
with this https://github.com/afedchin/xbmc/commit/39898904ee90a6866b20d122cf025d81423dfaf5 I have correct aspect for anamorph and full stereo. as I said above I will test with more samples tonight.

**Comment by FernetMenta** [comment]
your sample does not have stereo_mode set by ffmpeg. stereo_mode is set by CStereoscopicsManager::DetectStereoModeByString. 
this is all rather hacky and a can of worms ...
we have to keep things clear and separated.

**Comment by afedchin** [comment]
but we support both stereo flags from a container and file naming convention.

**Comment by FernetMenta** [comment]
but the one form file naming must not be used to determine aspect ratio. PR is updated.

**Comment by afedchin** [comment]
this https://www.dropbox.com/s/an8v07r1t3k0e9h/3d.Kodi.test.mkv?dl=0 sample (anamorph stereo) still not work as expected.

**Comment by FernetMenta** [comment]
@afedchin I looked into ffmpeg code and noticed that AR already observes stereo modes. All samples I got work with this change.

**Comment by afedchin** [comment]
after quick test seems it returns pre ffmpeg 3.x behavior

**Comment by FernetMenta** [comment]
great, thanks

jenkins build this please

**Pull request merged**